extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java

```diff
@@ -406,6 +411,49 @@ void builder(Optional<EffectiveIdeBuildItem> effectiveIdeBuildItem, BuildProduce
         }
     }
 
+    @BuildStep
+    BrowserOpenerBuildItem builder(HttpRootPathBuildItem rp, NonApplicationRootPathBuildItem np) {
+        return new BrowserOpenerBuildItem(new Consumer<String>() {
+            @Override
+            public void accept(String s) {
+                if (s.startsWith("/q")) {
+                    s = np.resolvePath(s.substring(3));
+                } else {
+                    s = rp.resolvePath(s.substring(1));
+                }
+
+                StringBuilder sb = new StringBuilder("http://");
+                Config c = ConfigProvider.getConfig();
+                sb.append(c.getOptionalValue("quarkus.http.host", String.class).orElse("localhost"));
+                sb.append(":");
+                sb.append(c.getOptionalValue("quarkus.http.port", String.class).orElse("8080"));
+                sb.append(s);
+                String url = sb.toString();
+
+                Runtime rt = Runtime.getRuntime();
+                OS os = OS.determineOS();
+                try {
+                    switch (os) {
```

Comment:
Isn't easier to just do `Desktop.getDesktop().browse(url)`?

Reply:
I tried it, it was really slow on my machine, slow enough that you press the key again because you think nothing is happening and then 15s later you end up with 5 windows open. Also it seemed to loose the URL. 

This was on linux, maybe it works better on other platforms.